frontend/src/components/modules/Tournaments/MatchButton.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useState } from "react";
+import {
+  Button,
+  Box,
+  Typography,
+  TextField,
+  Modal,
+  Select,
+  MenuItem,
+  FormControl,
+  InputLabel,
+  type ButtonProps
+} from "@mui/material";
+import type { TournamentPlayer } from "./OngoingTournament/RoundRobin/RoundRobinTournamentView";
+import type { Match, Tournament } from "types/models";
+import PlayerName from "./PlayerNames";
+import api from "api/axios";
+import type { ChangeCourtTimeRequest } from "types/requests";
+
+interface MatchButtonProps {
+  match: Match;
+  players: TournamentPlayer[];
+  navigate: (path: string) => void;
+  t: (key: string) => string;
+  haveSameNames: boolean;
+  props: ButtonProps;
+  isUserTheCreator: boolean;
+  tournamentData: Tournament;
+}
+
+const MatchButton: React.FC<MatchButtonProps> = ({
+  match,
+  players,
+  navigate,
+  t,
```

Comment:
What does the t means here?

Reply:
ah yes good catch the t is the translation it was needed when the button was in function but now could be imported straight to the component. I will fix this.